sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/display/DisplayData.java

```diff
@@ -345,6 +373,14 @@ ItemBuilder addIfNotDefault(
      * <p>Leaving the namespace unspecified will default to the registering instance's class.
      */
     ItemBuilder withNamespace(Class<?> namespace);
+
+    /**
+     * Adds an explicit namespace to the most-recently added display metadata. The namespace
+     * and key uniquely identify the display metadata.
+     *
+     * <p>Leaving the namespace unspecified will default to the registering instance's class.
+     */
+    ItemBuilder withNamespace(JavaClass namespace);
```

Comment:
I was stewing on this a lot last night. I don't really like having an "options" object that receives extra configuration because we're really just setting fields on `DisplayData.Item`. It's weird that the API for configuring key + value would be so separate from other properties.

I'm thinking more about an API that looks like `ImmutableList.builder()`, where you're more explicitly putting built `DisplayData.Item`'s into a collection. Something like:

    builder
        .put(DisplayData.item("foo1", "bar1").withLabel("Foo 1"))
        .put(DisplayData.item("foo2", "bar2")
            .withNamespace("blarg")
            .withLinkUrl("http://goog"))
        .put("foo3", "bar3");

I've added an item to track this [[BEAM-199]](https://issues.apache.org/jira/browse/BEAM-199) so we can proceed with this PR.